src/test/java/io/cdap/plugin/sfmc/source/MarketingCloudInputFormatTest.java

```diff
@@ -0,0 +1,166 @@
+                                                                       "Data Extension,Unsub Event",
+                                                                       "dataExtensionKeys1,dataExtensionKey2",
+                                                                       "tableNameField",
+                                                                       "filter", CLIENT_ID, CLIENT_SECRET,
+                                                                       "restEndPoint", AUTH_ENDPOINT, SOAP_ENDPOINT);
+    SourceQueryMode mode = SourceQueryMode.MULTI_OBJECT;
+    SourceObject object = SourceObject.DATA_EXTENSION;
+    ETConfiguration conf = Mockito.spy(new ETConfiguration());
+    conf.set("clientId", CLIENT_ID);
+    conf.set("clientSecret", CLIENT_SECRET);
+    conf.set("authEndpoint", AUTH_ENDPOINT);
+    conf.set("soapEndpoint", SOAP_ENDPOINT);
+    conf.set("useOAuth2Authentication", "true");
+    List<SourceObject> list = new ArrayList<>();
+    SourceObject object1 = SourceObject.DATA_EXTENSION;
+    SourceObject object2 = SourceObject.TRACKING_UNSUB_EVENT;
+    list.add(object1);
+    list.add(object2);
+    List<MarketingCloudColumn> columns = new ArrayList<>();
+    MarketingCloudColumn column1 = new MarketingCloudColumn("sys_created_by", "string");
+    MarketingCloudColumn column2 = new MarketingCloudColumn("sys_updated_by", "string");
+    columns.add(column1);
+    columns.add(column2);
+    MarketingCloudObjectInfo info = new MarketingCloudObjectInfo(object, columns);
+    List<MarketingCloudObjectInfo> tableInfos = new ArrayList<>();
+    tableInfos.add(info);
+    ETClient etClient = PowerMockito.mock(ETClient.class);
+    PowerMockito.whenNew(ETClient.class).withArguments(Mockito.anyString()).thenReturn(etClient);
+    PowerMockito.spy(new MarketingCloudClient(new ETClient(conf)));
+    Mockito.when(config.getObjectList()).thenReturn(list);
+    Configuration jobConfig = Mockito.mock(Configuration.class);
+    PowerMockito.spy(MarketingCloudInputFormat.class);
+    PowerMockito.when(MarketingCloudInputFormat.fetchTableInfo(mode, config)).thenReturn(tableInfos);
+    MarketingCloudInputFormat.setInput(jobConfig, mode, config);
+    Assert.assertEquals(1, MarketingCloudInputFormat.fetchTableInfo(mode, config).size());
+  }
+
+  @Test
+  public void testCreateRecordReader() throws Exception {
+    InputSplit inputSplit = Mockito.mock(InputSplit.class);
+    TaskAttemptContext taskAttemptContext = Mockito.mock(TaskAttemptContext.class);
+    MarketingCloudJobConfiguration jobConfiguration = Mockito.mock(MarketingCloudJobConfiguration.class);
+    PowerMockito.whenNew(MarketingCloudJobConfiguration.class).withArguments(Mockito.any())
+      .thenReturn(jobConfiguration);
+    MarketingCloudInputFormat marketingCloudInputFormat = new MarketingCloudInputFormat();
+    marketingCloudInputFormat.createRecordReader(inputSplit, taskAttemptContext);
```

Comment:
this test is not validating anything

Reply:
Added the assertion for it to test that the recordReader created is notNull.